github/github.go

```diff
@@ -542,6 +569,25 @@ func (r RateLimits) String() string {
 	return Stringify(r)
 }
 
+type rateLimitCategory uint8
+
+const (
+	coreCategory rateLimitCategory = iota
+	searchCategory
+
+	categories // An array of this length will be able to contain all rate limit categories.
+)
+
+// category returns the rate limit category of the endpoint, determined by Request.URL.Path.
+func category(path string) rateLimitCategory {
+	switch {
+	default:
```

Comment:
It's a style I've iterated towards after using `switch` statements more for multiplexing across multiple branching decisions. I prioritize the order of cases, returning `coreCategory` and `searchCategory`, to be more consistent with how they're defined, over the conditions that lead to those return values.

I read it as:

> For all paths, return `coreCategory`, except when it has `/search/` prefix, then return `searchCategory`.

I personally like it because I think the cases themselves are more important than the conditions to trigger them, but I know this is less common. If people feel this is worse, I can revert to a more traditional style of sorting by conditions (putting `default` last).